parsing/src/processor.ts

```diff
@@ -67,8 +67,15 @@ function removeRubbishItems(items: PartDataInterface, recipes: Recipe[]): void {
 }
 
 // Central function to process the file and generate the output
-async function processFile(inputFile: string, outputFile: string) : Promise<{ buildings: { [key: string]: number }; items: PartDataInterface; recipes: Recipe[] } | undefined> {
-    try {
+async function processFile(
+    inputFile: string, 
+    outputFile: string) : Promise<{ buildings: { 
```

Comment:
Indentation mega jank here.

Recommend you create a new interface and use that interface type in here instead.

E.g. `interface OutputFileInterface = { ... all the things }`

Then:

```suggestion
    outputFile: string) : Promise<OutputFileInterface | undefined>
```

Reply:
I didn't touch this. Deferring this suggestion for now.